javademo/src/main/java/com/example/javademo/Main.java

```diff
@@ -31,14 +35,21 @@ public static void main(String[] args) throws Exception {
         Path contractPath = Paths.get("balance.cairo");
         String contract = String.join("", Files.readAllLines(contractPath));
 
-        // Deploy a contract
+        // Declare a contract
+        // Class hash has to be calculated manually
         ContractDefinition contractDefinition = new ContractDefinition(contract);
-        DeployTransactionPayload payload = new DeployTransactionPayload(contractDefinition, Felt.fromHex("0x1234"), Collections.emptyList(), Felt.ZERO);
-        Request<DeployResponse> deployRequest = provider.deployContract(payload);
-        DeployResponse deployResponse = deployRequest.send();
+        Felt classHash = Felt.fromHex("0x399998c787e0a063c3ac1d2abac084dcbe09954e3b156d53a8c43a02aa27d35");
+        Felt maxFee = Felt.ZERO;
+        Felt nonce = account.getNonce().send();
+        ExecutionParams executionParams = new ExecutionParams(nonce, maxFee);
+        DeclareTransactionPayload declareTransactionPayload = account.signDeclare(contractDefinition, classHash, executionParams);
+
+        // Deploy a contract with Universal Deployer Contract
+        StandardDeployer contractDeployer = new StandardDeployer(Felt.fromHex("0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02bf"), provider, account);
+        ContractDeployment deployResponse = contractDeployer.deployContract(classHash, true, Felt.fromHex("0x12345678"), emptyList()).send();
```

Comment:
Same here, lets add `....send()` and also I'd show how to use a `Deployer#findContractAddress` to get the address of the deployed contract.

Reply:
> Same here, lets add ....send()
Not sure I follow, this request is already sent in the same line. Can you elaborate?

Reply:
I missed the `.send` at the at the end of the line. Disregard that.

I'd still show findContractAddress usage though.